### DISCUSSION TITLE
Temporary address lockdown feature

### DISCUSSION THREAD

**Discussion by nghuyenthevinh2000** [classic-terra/core#202] [Ideas]
# Temporary address lockdown feature

## Introduction
Due to recent exploit on Terraport, a need for a lockdown feature is required to temporarily disable sending on suspected addresses. This temporary lockdown will allow teams to investigate and respond carefully.

## Design

A new module "security" will be created to address this issue. Here is the design for this "security" module:

1. Parameters:
* LockdownTime (second): 1209600 (14 days)
* LockdownIssuer (address): only one allowed multisig addresses can invoke LockDown.

2. Transaction
* LockDown: lock an array of addresses for a LockdownTime period. Only LockdownIssuer can invoke this LockDown logic. This logic will put addresses in to a KV store.

3. Ante handler
* A custom ante handler will be put in place to check for sender of MsgSend, MsgMultisend. If it detects lockdown addresses then the Msg will be discarded.
* This custom ante handler will be after gas fee is collected.

**Comment by faddat** [comment]
This hasn't been approved by governance.

**Reply by nghuyenthevinh2000** [reply]
Yes, it is true. I am waiting for validator opinions on this
